railties/lib/rails/commands/statistics/statistics_command.rb

```diff
@@ -26,12 +26,21 @@
   %w(Channel\ tests     test/channels),
   %w(Integration\ tests test/integration),
   %w(System\ tests      test/system),
-].collect do |name, dir|
-  [ name, "#{File.dirname(Rake.application.rakefile_location)}/#{dir}" ]
-end.select { |name, dir| File.directory?(dir) }
+]
 
-desc "Report code statistics (KLOCs, etc) from the application or engine"
-task :stats do
-  require "rails/code_statistics"
-  CodeStatistics.new(*STATS_DIRECTORIES).to_s
+module Rails
+  module Command
+    class StatisticsCommand < Base # :nodoc:
+      desc :stats, "Report code statistics (KLOCs, etc) from the application or engine"
+      def perform
+        require "rails/code_statistics"
```

Comment:
previously we used the following piece of code in the `railties/lib/rails/tasks.rb` 

```
tap { |arr|
  arr << "statistics" if Rake.application.current_scope.empty?
}
```
Do you think this is a good equivalent of what it did?

```suggestion
        return if Rake.application.current_scope.any?
        
        require "rails/code_statistics"
```

Reply:
I don't think that is still a problem when it's no longer a Rake task.
If you look at the commit that added it, it seems related to engines loading the task twice.
As it is now a Thor task I don't think it will be loaded twice.
https://github.com/rails/rails/commit/38f28f77049e7160491b71167471a2175253b742

Reply:
That makes sense, thank you!